enhancements/windows-containers/bring-your-own-windows-host.md

```diff
@@ -0,0 +1,171 @@
+Windows instances to an OpenShift 4.7+ cluster using the Windows Machine Config
+Operator (WMCO).
+
+## Motivation
+
+The world of Windows sees customers treating their instances more as "pets"
+rather than "cattle". There is a desire from customers to be able to reuse these
+"pet" Windows instances as OpenShift worker nodes, run Windows workloads and gain
+similar benefits that their Linux workloads get when being managed by OpenShift.
+
+### Goals
+
+* Allow a customer to add an existing Windows instance running a variant of
+  Windows Server 2019 (1809, 1909, 20H2 etc). The instance should have the
+  Docker runtime and SSH services configured. The public key used by WMCO for
+  the [IPI workflow](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator.md#design-details)
+  needs to be an authorized key in the instance.
+* WMCO will then perform all the required steps within the VM for it to be
+  added to the cluster as an OpenShift worker node and removed from it.
+* [Handle upgrades](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+  of all components installed by WMCO.
+
+### Non-Goals
+
+* Installing the container runtime in the Windows instance
+* Handle upgrading of the Windows operating system
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The cluster admin will create a ConfigMap called _windows-instances_ in the
+_openshift-windows-machine-config-operator_ namespace. WMCO will be configured
+to watch for this special ConfigMap, whose data section will have the following
+layout:
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: windows-instances
+  namespace: openshift-windows-machine-config-operator
+data:
+  instance.1: |-
+    username=Administrator
+    ip-address=10.1.42.1
+  instance.2: |-
+    username=core
+    ip-address=10.1.42.2
+```
+WMCO will configure or de-configure instances based on the contents of the data
+section and will use the `cloud-private-key` to communicate with them.
+
+### User Stories
+
+#### Story 1
+As an OpenShift cluster administrator I would like to add an existing Windows
+instance to the cluster as a worker node.
+
+#### Story 2
+As an OpenShift cluster administrator I would like to remove an existing Windows
+instance from the cluster as a worker node.
+
+#### Story 3
+As an OpenShift cluster administrator I would like to have the Kubernetes
+components on Windows nodes upgraded when new Kubernetes versions or patch
+versions are released.
+
+### Risks and Mitigations
+
+The same risk and mitigation that apply to Windows nodes added using MachineSets
+to an IPI cluster applies here.
+
+## Design Details
+
+Changes to the _windows-instances_ ConfigMap will cause WMCO to compare the
+list of instances with the set of existing Windows nodes. The comparison will
```

Comment:
I dont see a mention of a controller here. It might be implied that there will be a new ConfigMap controller. Might be worth calling it out

Reply:
thought: A good portion of this work might be making the Machine controller logic usable by the configmap controller

Reply:
I agree. I also want to see a few more sentences on what WMCO controller does in the following scenarios? 
- Will the cluster be a combination of Windows Machines + BYOH Windows  VMs? 
- Or when WMCO is running in BYOH mode the WMCO controller is simply not started assuming there are no Windows Machines supported in this  scenario.

Reply:
I can add a few lines about a controller.

>thought: A good portion of this work might be making the Machine controller logic usable by the configmap controller

Correct but I don't need to go that deep into the implementation detail here.

>  Will the cluster be a combination of Windows Machines + BYOH Windows VMs?

Yes there is potential for this given UPI clusters have the Machine CRD and required controllers present.

> Or when WMCO is running in BYOH mode the WMCO controller is simply not started assuming there are no Windows Machines supported in this scenario.

I view this as implementation details. For starters say the Machine CRD is not available on the cluster, we should gracefully handle that.